common/djangoapps/course_category/models.py

```diff
@@ -83,6 +83,27 @@ def save_reindex_course_category(sender, instance, pk_set, action, **kwargs):
         instance.pre_clear_course_keys.update(str(x.id) for x in instance.courses.all())
 
     if action in ['post_add', 'post_remove']:
-        courses_set.update(str(x.id) for x in instance.courses.all())
+        courses_set.update(str(c.id) for c in instance.courses.all())
         courses_set.update(getattr(instance, 'pre_clear_course_keys', set()))
         task_reindex_courses.delay(instance.id, list(courses_set))
+
+
+@receiver(post_init, sender=CourseCategory)
+def remember_previous_name(sender, instance, **kwargs):
+    """
+    Saves category name to check if it was changed.
+    """
+    instance.previous_name = instance.name
+
+
+@receiver(post_save, sender=CourseCategory)
+def category_name_change_reindex_course_category(sender, instance, created, **kwargs):
+    """
+    Reindexes category courses when category name changes.
+
+    We need to reindex category courses (in case of category name was changed but not category courses)
```

Comment:
```suggestion
    We need to reindex category courses (in case of category name was changed but not category courses)
```
```suggestion
    We need to reindex category courses.
    In case of a category name was changed but not category courses).
```